sandbox/populate_maturity_level_in_es_yindex.py

```diff
@@ -0,0 +1,65 @@
+"""Populate maturity-level in yindex"""
+
+import json
+import os
+
+from elasticsearchIndexing.es_manager import ESManager
+from elasticsearchIndexing.models.es_indices import ESIndices
+from redisConnections.redisConnection import RedisConnection
+from utility import log
+from utility.create_config import create_config
+
+ES_CHUNK_SIZE = 100
+
+
+def main():
+    config = create_config()
+    logger = log.get_logger(
+        'populate_maturity_level_yindex',
+        os.path.join(config.get('Directory-Section', 'logs'), 'sandbox.log'),
+    )
+    es_manager = ESManager()
+    modules_for_bulk_indexing = []
+    for module_key, module in all_modules_generator():
+        maturity_level = module.get('maturity-level')
+        if (
+            not maturity_level
+            or (maturity_level and maturity_level.lower() == 'none')
+            or not es_manager.document_exists(ESIndices.YINDEX, module)
+        ):
+            continue
+        logger.info(f'Module: {module_key} has such maturity-level: {maturity_level}')
+        query = {
+            'query': {
+                'bool': {
+                    'must': [
+                        {'term': {'module': module['name']}},
+                        {'term': {'organization': module['organization']}},
+                    ],
+                },
+            },
+        }
+        es_response = es_manager.generic_search(index=ESIndices.YINDEX, query=query, response_size=None)['hits']['hits']
+        for result in es_response:
+            module_data = result['_source']
+            module_data['maturity-level'] = maturity_level
+            modules_for_bulk_indexing.append(module_data)
+    logger.info(f'{len(modules_for_bulk_indexing)} modules are ready to be updated in Elasticsearch')
+    i = 0
+    while i <= len(modules_for_bulk_indexing):
+        chunk = modules_for_bulk_indexing[i : i + ES_CHUNK_SIZE]
+        es_manager.bulk_modules(ESIndices.YINDEX, chunk)
```

Comment:
Does this really just update the documents? I think this would add a second version of the document, so the one without the maturity-level will stay in the index as well. I don't think you can modify the source of document after it's been indexed. Either you need to reindex, or use the [update API](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html). I think the update API let's you set properties without modifying the source.

Reply:
thanks, changed to update API

Reply:
https://gist.github.com/Fagtoy/353c009b90369e017b3577eab103198f
this script was removed from the repo

Reply:
OK, you need to check the revision when setting rfc to true though. An older revision of a module might have been published in a draft and a later one in an rfc. The revision from the draft shouldn't be marked as ratified.

Reply:
> OK, you need to check the revision when setting rfc to true though. An older revision of a module might have been published in a draft and a later one in an rfc. The revision from the draft shouldn't be marked as ratified.

how can I check it correctly?

Reply:
Updated the [gist](https://gist.github.com/Fagtoy/353c009b90369e017b3577eab103198f)